java/code/src/com/redhat/rhn/domain/errata/ErrataFactory.java

```diff
@@ -197,50 +191,6 @@ else if (identifier.length() > 4) {
         return retval;
     }
 
-    /**
-     * publish takes an unpublished errata and copies its contents into a Published Errata
-     * object (and then returns this object). This method also handles removing the old
-     * Unpublished Errata object and child elements from the db.
-     * @param unpublished The Errata to publish
-     * @return Returns a published errata.
-     */
-    public static Errata publish(Errata unpublished) {
-        //Make sure the errata we're publishing is unpublished
-        if (unpublished.isPublished()) {
-            return unpublished; //there is nothing we can do here
-        }
-        //Create a published errata using unpublished
-
-        Errata published;
-
-        if (unpublished.isCloned()) {
-            published = new PublishedClonedErrata();
-            ((PublishedClonedErrata)published).setOriginal(
-                    ((UnpublishedClonedErrata)unpublished).getOriginal());
-        }
-        else {
-            published = ErrataFactory.createPublishedErrata();
-        }
-
-        copyDetails(published, unpublished, false);
-
-        //Save the published Errata
-        save(published);
-
-        //Remove the unpublished Errata from db
-        try {
-            Session session = HibernateFactory.getSession();
-            session.delete(unpublished);
-        }
-        catch (HibernateException e) {
-            throw new HibernateRuntimeException(
-                    "Errors occurred while publishing errata", e);
-        }
-
-        //return the published errata
-        return published;
-    }
-
     /**
      * Takes a published or unpublished errata and publishes to a channel, creating
```

Comment:
should the documentation be change? Now it will not publish an unpublish errata.
The same comment can be applied to the next methods.

Reply:
Yes, and it is indeed correct. Not sure why GitHub seems to display an outdated version of the file.